### PR TITLE
Fixed several issues in the notebook:

### PR DESCRIPTION
**Pull request by chorus12**
1. The code in main lycle error = error_formula(y, output) is declared and never used
2. error_foumula function is for a binary cross entropy but all the backprop and loss are for MSE
Added extended narrative for both BCE and MSE
Removed unused code
Removed division by n_features since it is compensated by the learning rate

### CODE REVIEW

**Comment by amit1rrr** [comment]
Sorry to intrude but I noticed this PR contains Jupyter notebook files that are hard to review on GitHub due to the JSON diffs.

You might want to checkout [ReviewNB](https://www.reviewnb.com/), code review tool for notebooks. Provides visual diffs & commenting for Jupyter notebooks on GitHub. Completely free for open source repositories.

Disclaimer: I built ReviewNB

diff_and_commenting
